ntbs-service/Pages/Notifications/Edit/ClinicalDetails.cshtml.cs

```diff
@@ -144,40 +145,59 @@ private void SetupNotificationSiteMap(IEnumerable<NotificationSite> notification
             }
         }
 
-        private IEnumerable<NotificationSite> CreateNotificationSitesFromModel(Notification notification)
+        private IEnumerable<NotificationSite> CreateNotificationSitesFromModel(int notificationId)
         {
             foreach (var item in NotificationSiteMap) {
                 if (item.Value) 
                 {
                     yield return new NotificationSite
                     {
-                        NotificationId = notification.NotificationId,
+                        NotificationId = notificationId,
                         SiteId = (int)item.Key,
                         SiteDescription = item.Key == SiteId.OTHER ? OtherSite.SiteDescription : null
                     };
                 }
             }
         }
 
-        public ContentResult OnGetValidateClinicalDetailsProperty(string key, string value)
+        public ContentResult OnGetValidateClinicalDetailsProperty(string key, string value, bool shouldValidateFull)
         {
-            return ValidateProperty(new ClinicalDetails(), key, value);
+            return ValidateModelProperty<ClinicalDetails>(key, value, shouldValidateFull);
         }
+
+        public ContentResult OnGetValidateClinicalDetailsListProperty(string key, IEnumerable<string> valueList, bool shouldValidateFull)
```

Comment:
What is this validation for? The name is confusing to me as it's clearly aobut notification sites in particular, but called "ClinicalDetailsListProperty" as if it was for a general list?

Reply:
This is for NotificationSites only, I have renamed it now